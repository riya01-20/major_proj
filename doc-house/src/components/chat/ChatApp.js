import React, {useEffect, useState} from 'react'
import "../../styles/App.css";
import Sidebar from './Sidebar';
import Chat from './Chat';
import axios from '../../axios';
import Navbar from '../general/Navbar';
import UserContext from '../../context/userContext';

function ChatApp() {
    const [user, setUser] = useState({});
    // console.log(user);

    useEffect(() => {
        const token = localStorage.getItem('auth-token');
        // console.log(token);
        const checkAuth = async () => {
            const userRes = await axios.post("/general/tokenIsValid",null, {
                headers: { "x-auth-token": token },
            });
            setUser({...userRes.data.result, type: userRes.data.type});
        }
        
        checkAuth();
    }, []);

    return (
        <UserContext.Provider value = {{user}}>
            <div> 
                <Navbar />
                <div className="app"> 
                    <div className="app__body">
                        <Sidebar/>
                        <Chat/>
                    </div>
                </div>
            </div>
        </UserContext.Provider>
    )
}

export default ChatApp
