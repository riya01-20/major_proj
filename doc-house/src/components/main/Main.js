import React from 'react';
import {Router,withRouter, Switch, Route } from 'react-router-dom';
import { createBrowserHistory } from "history";
import HomePage from '../general/HomePage';
import LoginPageUser from '../user/LoginPageUser';
import LoginPageDoctor from '../doctor/LoginPageDoctor';
import LogTransition from '../general/LogTransition';
import DoctorDashboard from '../doctor/DoctorDashboard';
import UserDashboard from '../user/UserDashboard';
import UserProfile from '../user/UserProfile';
import DoctorProfile from '../doctor/DoctorProfile';
import SignUpPageUser from '../user/SignUpPageUser';
import SignUpPageDoctor from '../doctor/SignUpPageDoctor';
import NotFound from '../general/NotFound';
import SearchDoctor from '../general/SearchDoctor';
import ViewProfile from '../general/ViewProfile';
import ChatApp from '../chat/ChatApp';

export const history = createBrowserHistory();

const Root = () => (
  <Switch>
    <Route exact path="/" component={HomePage}/>
    <Route path="/user-dashboard" component={UserDashboard}/>
    <Route path="/doctor-dashboard" component={DoctorDashboard}/>
    <Route path="/log-transition" component={LogTransition}/>
    <Route path="/signup-user" component={SignUpPageUser}/>
    <Route path="/signup-doctor" component={SignUpPageDoctor}/>
    <Route path="/login-user" component={LoginPageUser}/>
    <Route path="/login-doctor" component={LoginPageDoctor}/>
    <Route path="/user-profile" component={UserProfile}/>
    <Route path="/doctor-profile" component={DoctorProfile}/>
    <Route path="/profile/:id" component={ViewProfile}/>
    <Route path="/search-doctor" component={SearchDoctor}/>
    <Route exact path="/chat" render={(props) => <ChatApp {...props} />} />
    <Route path="/chat/rooms/:roomId" render={(props) => <ChatApp {...props} />} />
    <Route component={NotFound}/>
  </Switch>
)

export const AppWithRouter = withRouter(Root)

const Main = () => (
  <Router history={history}>
    <AppWithRouter/>
  </Router>
)

export default Main;