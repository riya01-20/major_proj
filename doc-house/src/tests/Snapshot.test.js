import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'
import Main from '../components/main/Main';
import ChatApp from '../components/chat/ChatApp';



describe('Snapshots', () => {
    it("renders correctly", () => {
        const tree = shallow(<Main />);
        expect(toJSON(tree)).toMatchSnapshot();
    });

    it("renders correctly", () => {
        const tree = shallow(<ChatApp />);
        expect(toJSON(tree)).toMatchSnapshot();
    });
});