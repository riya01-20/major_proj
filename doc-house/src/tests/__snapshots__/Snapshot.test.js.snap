// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(Root) />
</Router>
`;

exports[`Snapshots renders correctly 2`] = `
<ContextProvider
  value={
    Object {
      "user": Object {},
    }
  }
>
  <div>
    <WithStyles(Navbar) />
    <div
      className="app"
    >
      <div
        className="app__body"
      >
        <Sidebar />
        <Chat />
      </div>
    </div>
  </div>
</ContextProvider>
`;
